### DISCUSSION TITLE
Security button

### DISCUSSION THREAD

**Discussion by wrwrabbit** [wrwrabbit/Partisan-Telegram-Android#99] [Ideas]
Новые идеи от @huntmix

**Comment by zalexdev** [comment]
Приветствую, я разработчик [Security addon](https://github.com/huntmix/securityaddon), приложение, которое имеет множество функций в плане защиты данных и самого девайса. На данный момент приложение ждет крупное обновление 15 мая, так что буду говорить о новой версии _(если требуется код - с радостью предоставлю)._
1. Шифрование сообщений и файлов с помощью AES перед отправкой (допустим docx, txt, png) - зажимаем по файлу вводим желаемый пароль, отсылается уже надежно зашифрованный файл, при выходе из чата пароль удаляется. При получении файла .aes при клике просит пароль, расшифровка, открытие из временной памяти файла, после закрытии удаляется (при желании). Пароль можно передавать при личной встрече  или секретном чате. Сообщения — зажимаем отправить и видим новый подпункт — зашифровать. Расшифровка так же как и с файлом. **(код предоставлю)**
2. Снятия на фото силовика и т.д. - кто пытается разблокировать телефон, и\или ввел фейк пароль в тг. Сделать можно через администратора устройства. Фиксируется дата и время с точностью до секунды, после сохранения можно так же будет отправлять доверенным лицам **(код предоставлю)**
3. Небольшой антивор который отправит смс с геолокацией по команде, или заблокирует\полностью сбросит устройство (тоже администратор устройства)
4. Фейковый ярлык, своя иконка, название и запуск приложения (допустим как вайбер, или тг) при котором идет запуск указанного приложения, но начинаются действия выбранные заранее **(код предоставлю)**
5. Отправка смс сообщение контактам (вдруг инета нет) **(код предоставлю)**
6. Удаление заранее выбранных файлов и папок (можно и шифрование) как действие при н случаи (можно активировать допустим через кодовое слово) **(код предоставлю)**
Жду ваш фидбек)

**Reply by vivabelarus** [reply]
Добрый день. Не до конца понял, что Вы предлагаете, но звучит интересно. Я так понимаю, наше приложение должно будет интегрироваться с вашим (если оно установлено) и использовать его функции?

1. Подумаем над шифрованием файлов. Не уверен, что подобная функция необходима/приоритетна. В шифровании сообщений тем более не вижу смысла. Но, может, чего-то не понимаю.
2. Супер! Это то, что нам нужно. В недавней версии мы уже реализовали логирование неудачных попыток входа. Фото силовика было бы логичным продолжением.
3. Отправка смс уже есть, но можем посмотреть. Может, у вас функционал лучше.
4. Я так понимаю, будет иконка тг, открывается тг, но при этом запускаются какие-то дополнительные действия? Как это можно было бы использовать?
5. Экстренная отправка смс? У нас уже есть такой функционал.
6. Да, удаление файлов нам не помешало бы.

Кроме того, мы ведём разработку [P-SMS](https://github.com/wrwrabbit/Partisan-SMS) (форк [QKSMS](https://github.com/moezbhatti/qksms)). Основной функционал приложения: зашифрованное общение при помощи смс и удаление смс по таймеру. Поэтому можем обсудить интеграцию и в рамках этого приложения.

**Reply by zalexdev** [reply]
Вообще речь идет даже не о интеграции, а о дополнительном функционале который можно взять с моей разработки. 
1. Допустим файл силовики.pdf за это можно и приплести статью, если же файл будет зашифрован - по сути это тупой набор  симоволов
2. Можно сделать как и при пинкоде от тг так и от самого девайса
3.  Смс по типу: родственник /locate - сообщение с гео, /lock - блокировка устройства, /photo - фото с фронталки
4.  Допустим вайбер, там тоже есть чаты но партизанского вайбера нет, делается ярлык как вайбер, и юзер будет знать что произойдет если нажать, а левый чел сразу нажмет тем самим запсукаю цепочку сос
5.  Недоглядел)
6. Отлично))

**Reply by vivabelarus** [reply]
1. Я так понимаю, Вы не из Беларуси и не слишком понимаете деталей нашей ситуации, поэтому я немного объясню. Дело в том, что у нас закон не работает совсем. Это означает, что, если человека хотят посадить, его посадят независимо от того, найдут у него что-то или нет. А с другой стороны, если каратели посчитают, что в .aes файлах есть то, что нужно им, они могут выбить из человека пароль от файла силой. Поэтому решения "просто зашифровать, они ж не расшифруют" не очень эффективны. Нужны решения из области стеганографии. Так, чтобы можно было строить из себя дурачка и говорить "да, это просто pdf с котиками, в нём нет скрытых экстремистских материалов".
3. А, не прочитал "по команде". То есть, например, в личку приходит команда /photo, а в ответ присылается фотка?
4. Ну вайбер да, не помешало бы. А тогда при чём здесь Partisan Telegram? Или внедрить этот функционал в него, чтобы не ставить отдельное приложение?

**Reply by zalexdev** [reply]
Вообще внедрить все выше указанные функции в п телеграмм, про /photo вы правильно поняли

**Reply by vivabelarus** [reply]
Последнее сообщение не заметил. Гитхаб его свернул. Да, можем внедрять. Вы как планируете, просто предоставить нам код? Или были бы не против поучаствовать в разработке?

**Reply by zalexdev** [reply]
Пока просто дать код, помочь если будут с ним проблемы, не знаю точно. Сам работаю над secaddon, возможно после 8 мая если что то не получится присоединюсь

**Reply by vivabelarus** [reply]
Окей, я тогда позже создам issues под каждую идею

**Reply by zalexdev** [reply]
Окей буду ждать

**Reply by vivabelarus** [reply]
Добрый день. Создал для этих пунктов issues с соответствующими метками. Только я не понял, как пункт 3 может быть реализован в рамках Partisan Telegram, поэтому для него не создавал.

Можете там оставить ссылки на код. И там же в случае необходимости можем обсуждать детали реализации.

**Reply by zalexdev** [reply]
Залил только что обновление на гит
